File: src/com/aslan/config/lookup.js
import Config from "./config.js";

export default class Lookup {

    static _invoker = null;
    static get invoker() {
        if (Lookup._invoker === null) {
            let Class = require('../module/command/ImmediateInvoker');
            Lookup._invoker = new Class.default();
        }
        return Lookup._invoker;
    };

    static _notification = null;
    static get notification() {
        if (Lookup._notification === null) {
            let Class = require('../module/notify/HashClassifyNotification');
            Lookup._notification = new Class.default();
        }
        return Lookup._notification;
    };

    static _seedSourceFactory = null;
    static get seedSourceFactory() {
        if (Lookup._seedSourceFactory === null) {
            let Class = require('../module/random/SourceSeedFactory');
            Lookup._seedSourceFactory = new Class.default();
        }
        return Lookup._seedSourceFactory;
    };
}

File: src/com/aslan/config/Lookup.java
package com.aslan.config;

import com.aslan.module.command.ImmediateInvoker;
import com.aslan.module.command.Invoker;
import com.aslan.module.notify.HashClassifyNotification;
import com.aslan.module.notify.Notification;
import com.aslan.module.random.SeedFactory;
import com.aslan.module.random.SourceSeedFactory;

public class Lookup {

    static Invoker _invoker = null;

    public static Invoker invoker() {
        if (Lookup._invoker == null) {
            Lookup._invoker = new ImmediateInvoker();
        }
        return Lookup._invoker;
    }

    static Notification _notification = null;

    public static Notification notification() {
        if (Lookup._notification == null) {

            Lookup._notification = new HashClassifyNotification();
        }
        return Lookup._notification;
    }

    static SeedFactory _seedSourceFactory = null;

    public static SeedFactory seedFactory() {
        if (Lookup._seedSourceFactory == null) {
            Lookup._seedSourceFactory = new SourceSeedFactory();
        }
        return Lookup._seedSourceFactory;
    }

}


